src/cast_core/src/MonotonicSingleLoader.cpp

```diff
@@ -82,15 +83,22 @@ void MonotonicSingleLoader::run() {
                     // Use ordered:false to increase write parallelism for sharded collections.
                     auto options = mongocxx::options::insert();
                     options.ordered(false);
-                    auto result = config->collection.insert_many(std::move(docs), options);
-
-                    totalOpCtx.addBytes(numBytes);
-                    individualOpCtx.addBytes(numBytes);
-                    if (result) {
-                        totalOpCtx.addDocuments(result->inserted_count());
-                        individualOpCtx.addDocuments(result->inserted_count());
+                    try {
+                        auto result = config->collection.insert_many(std::move(docs), options);
+                        totalOpCtx.addBytes(numBytes);
+                        individualOpCtx.addBytes(numBytes);
+                        if (result) {
+                            totalOpCtx.addDocuments(result->inserted_count());
+                            individualOpCtx.addDocuments(result->inserted_count());
+                        }
+                    } catch (const mongocxx::operation_exception& e) {
+                        individualOpCtx.failure();
+                        totalOpCtx.failure();
+                        BOOST_LOG_TRIVIAL(info)
+                            << " Error inserting: " << bsoncxx::to_json(e.raw_server_error().get());
```

Comment:
Surprisingly to me, if you have no mongod running, you get the error message:
```
 Error inserting: {}
```

This seems like something weird about the C++ driver?

Reply:
Hm. Weird indeed. Not sure what to suggest, but we can file a ticket with the drivers team if we'd like to pursue it.